File: python/burkut/resources/gold.py
"""
Gold and Precious Metals Resource
"""

from typing import Any, Dict, List
from burkut.resources.base import BaseResource


class GoldResource(BaseResource):
    """Altın ve kıymetli maden fiyatları (Gram Altın, Çeyrek Altın, Ons vb.)."""

    def list(self) -> List[Dict[str, Any]]:
        """
        Tüm altın türlerini listeler.

        Returns:
            Altın türleri ve fiyatları listesi.
        """
        res = self._get("/gold")
        return res if isinstance(res, list) else []

    def get(self, symbol: str) -> Dict[str, Any]:
        """
        Belirli bir altın türünün fiyat bilgisini getirir.

        Args:
            symbol: Altın kodu (Örn: 'ALTIN_GRAM', 'ALTIN_CEYREK', 'ONS')
        """
        if not symbol or not isinstance(symbol, str):
            raise ValueError("symbol parametresi boş olamaz.")
        return self._get(f"/gold/{symbol.strip().upper()}")

File: nodejs/src/resources/gold.ts
import { BaseResource } from './base';
import type { GoldItem } from '../types';

export class GoldResource extends BaseResource {
  /**
   * Tüm altın türlerini ve güncel fiyatlarını listeler.
   */
  async list(): Promise<GoldItem[]> {
    return this._get<GoldItem[]>('/gold');
  }

  /**
   * Belirli bir altın türünün fiyat bilgisini getirir.
   * @param symbol Altın kodu (Örn: 'ALTIN_GRAM', 'ALTIN_CEYREK', 'ONS')
   */
  async get(symbol: string): Promise<GoldItem> {
    if (!symbol || typeof symbol !== 'string') {
      throw new Error('symbol parametresi geçerli bir metin olmalıdır.');
    }
    return this._get<GoldItem>(`/gold/${symbol.trim().toUpperCase()}`);
  }
}
